src/snowflake/cli/plugins/nativeapp/utils.py

```diff
@@ -81,3 +81,33 @@ def verify_exists(paths_to_sync: List[Path]):
     for path in paths_to_sync:
         if not path.exists():
             raise ClickException(f"The following path does not exist: {path}")
+
+
+def get_all_file_paths_under_dir(directory: Path):
+    """
+    Recursively traverse the given directory and yield file paths. If the directory is a symlink, then the function follows the behavior of os.walk for symlinks.
+    The directory may be a relative path to the project root.
+    """
+    for root, _, files in os.walk(directory):
+        for file in files:
+            yield Path(os.path.join(root, file))
+
+
+def is_python_file(file_path: Path):
+    """
+    Checks if the given file is a python file.
+    """
+    return file_path.suffix == ".py"
```

Comment:
nit: this doesn't need to be a general utility (or not yet), I'd keep it localized to the logic that uses it.